lib/presentation/widget/myspace/my_space_widget.dart

```diff
@@ -123,18 +124,22 @@ class _MySpaceWidgetState extends State<MySpaceWidget> {
   }
 
   Widget _buildMySpaceList(BuildContext context) {
-    return StoreConnector<AppState, List<Document>>(
-        converter: (store) => store.state.mySpaceState.documentList,
-        builder: (context, documentList) {
-          if (documentList.isEmpty) {
-            return _buildUploadFileHere(context);
-          } else {
-            return ListView.builder(
-                itemCount: documentList.length,
-                itemBuilder: (context, index) {
-                  return _buildMySpaceListItem(context, documentList[index]);
-                });
-          }
+    return StoreConnector<AppState, MySpaceState>(
+        converter: (store) => store.state.mySpaceState,
+        builder: (context, mySpaceState) {
+            return (mySpaceState.viewState.any((viewState) => viewState is LoadingState)) ?
+                _buildMySpaceListView(context, mySpaceState.documentList) :
+                RefreshIndicator(
+                    onRefresh: () async => mySpaceViewModel.getAllDocument(),
+                    child: _buildMySpaceListView(context, mySpaceState.documentList));
+        });
+  }
+
+  Widget _buildMySpaceListView(BuildContext context, List<Document> documentList) {
+    return ListView.builder(
+        itemCount: documentList.length,
+        itemBuilder: (context, index) {
+          return documentList.isEmpty ? _buildUploadFileHere(context) : _buildMySpaceListItem(context, documentList[index]);
```

Comment:
This is an `itemBuilder` for `ListView`, You should not `_buildUploadFileHere` in here. that should be out side of `ListView.builder`. Please find another way to do this :(

Reply:
My bad forgot that. I put it outside but still in the function is it ok ?